src/wallet/wallet2.cpp

```diff
@@ -6642,11 +6642,23 @@ void wallet2::get_outs(std::vector<std::vector<tools::wallet2::get_outs_entry>>
         }
 
         // while we still need more mixins
+        uint64_t num_usable_outs = num_outs;
+        bool allow_blackballed = false;
         while (num_found < requested_outputs_count)
         {
           // if we've gone through every possible output, we've gotten all we can
-          if (seen_indices.size() == num_outs)
-            break;
+          if (seen_indices.size() == num_usable_outs)
+          {
+            // there is a first pass which rejects blackballed outputs, then a second pass
+            // which allows them if we don't have enough non blackballed outputs to reach
+            // the required amount of outputs (since consensus does not care about blackballed
+            // outputs, we still need to reach the minimum ring size)
+            if (allow_blackballed)
+              break;
+            MINFO("Not enough non blackballed outputs, we'll allow blackballed ones");
```

Comment:
Would it be helpful here to prompt the user if they want to continue w/ blackballed outputs? Or maybe at the end of output selection, display number of blackballed outputs used (if any)?

Reply:
If this happens, the output is only spendable if it uses blackballed outputs, so it'd be giving the user a choice of using or losing the monero.

Reply:
Ok, I misunderstood then. Thought the user could stop, and retry again when there might be more non-blackballed outputs available.

I can think of scenarios where losing it might be better than using with reduced privacy, the opposite is obvious.

Reply:
Well, there might be other outpputs from the wallet to choose from, but as it stand retrying would most likely just select the same ones. There would need to be a patch keeping track (in the wallet) of the amount of blackballed out per amount and not picking those in the first place. That'd mean they're unpendable, but that could be made a wallet setting.